third_party/move/move-prover/boogie-backend/src/options.rs

```diff
@@ -340,31 +363,73 @@ impl BoogieOptions {
         }
     }
 
-    fn check_version_is_greater(tool: &str, given: &str, expected: &str) -> anyhow::Result<()> {
+    fn check_version_comparable(
+        tool: &str,
+        given: &str,
+        expected_min: Option<&str>,
+        expected_max: Option<&str>,
+    ) -> anyhow::Result<()> {
+        if let Some(expected) = expected_min {
+            Self::check_version_op(tool, given, expected, "min")?;
+        }
+        if let Some(expected) = expected_max {
+            Self::check_version_op(tool, given, expected, "max")?;
+        }
+
+        Ok(())
+    }
+
+    fn check_part_is_geq(a_parts: Vec<&str>, b_parts: Vec<&str>) -> bool {
+        let mut result = true;
+        for (a, b) in a_parts.into_iter().zip(b_parts.into_iter()) {
+            let an = a.parse::<usize>().unwrap();
+            let bn = b.parse::<usize>().unwrap();
+            if an < bn {
+                result = false;
+                break;
+            }
+        }
+        result
+    }
+
+    fn check_version_op(tool: &str, given: &str, expected: &str, op: &str) -> anyhow::Result<()> {
```

Comment:
It is generally a bad idea in most languages to use a string (or string slice, here) to select different behavior from a function.  In Rust, it's easy enough to use an enum:
```
enum Op { Max, Min }
```
and then pass Op::Max or Op::Min instead of a string "max" or "min".

Though: to minimize code, since these really only affect the ordering used to compare parameters and the string output with the error case you can separate comparison parameters from output parameters and use a method with a signature like:
```
  fn check_version_le(
        expect_lesser: &str,
        expect_greater: &str,
        relative_term: &str,
        expected: &str,
        given: &str,
        tool: &str,
    ) -> anyhow::Result<()> {
        ...
        return Err(anyhow!("expected at {} version {} but found {} for `{}`",
            relative_term, expected, given, tool);
        ...
}
```
and call it with
```
check_version_le(min, given, "least", min, given, tool)
```
and
```
check_version_le(given, max, "most", max, given, tool)
```

Reply:
Thanks Brian. I have updated the function accordingly.